### PR TITLE
:sparkles: Add support for ImagePullSecrets in DeploymentSpec

### PR DESCRIPTION
**Pull request by adriananeci**
**What this PR does / why we need it**:

In order to be able to pull provider images from a private registry we need to be able to also pass the image pull secrets so that we can authenticate to the registry while pulling the image.

This PR adds support for `ImagePullSecrets` option in provider `DeploymentSpec`

### CODE REVIEW

**Review by alexander-demicev — APPROVED** [review]
thanks!
/approve

File: internal/controller/component_customizer.go [hunk 1]
```diff
@@ -110,6 +110,10 @@ func customizeDeployment(pSpec operatorv1.ProviderSpec, d *appsv1.Deployment) {
 			d.Spec.Template.Spec.Tolerations = dSpec.Tolerations
 		}
 
+		if dSpec.ImagePullSecrets != nil {
```
Comment (Fedosin) [inline internal/controller/component_customizer.go:113]: Probably the best way to deal with the lint issue is to add `//nolint:nestif` before the if block on line 94

**Comment by Fedosin** [comment]
Fix for the linter issue: https://github.com/kubernetes-sigs/cluster-api-operator/pull/176

**Review by Fedosin — APPROVED** [review]
/lgtm

**Comment by Fedosin** [comment]
@adriananeci thank you for working on this!

**Pull request merged**